### DISCUSSION TITLE
error in segment_trees()

### DISCUSSION THREAD

**Discussion by taeyoonlee87** [r-lidar/lidR#693] [Q&A] [answered]
Hello, 

I tried to segment individual trees. 

I created CHM using 'rasterize_canopy' and created the tree tops using 'locate_trees'. 

I applied "st_as_sf" to convert tree tops to SF objects. Like below 

```
opt_output_files(las_cat) <- ""
ttops <- locate_trees(las_cat, lmf(5), uniqueness = "bitmerge")
ttops2<- st_as_sf(ttops)
```

Then, I segment it using "segement_trees"

```
opt_output_files(las_cat) <- "/scratch/tl96575/mapping/chm/segmented_{XLEFT}_{YBOTTOM}"
algo <- silva2016(chm, ttops2)
ctg_segmented <- segment_trees(las_cat, algo)
```

but it resulted in the error message 

```
#Chunk 1 of 635 (0.2%): state ✗
#Error: On-disk rasters not supported in 'raster_as_dataframe()'
```

When I test my code with a small portion of data, it worked without errors. But when I ran my code for the whole dataset, it gave me this error. 

Did you ever see this error? Or could you please advise me on what point I need to check.?

**Comment by Jean-Romain** [comment, marked as answer]
Your code looks correct and should be handled automatically by `segment_trees()`. I will call it a bug.

#694

**Reply by taeyoonlee87** [reply]
Then, what can I do to get my results?

**Reply by Jean-Romain** [reply]
You can also process chunks of the point cloud and the raster yourself without using lidR in a loop. You sequentially crop and load 1x1 km of data in a loop. Alternatively you can try to fix the bug yourself. I'm currently in vacation and won't fix the bug until August.
